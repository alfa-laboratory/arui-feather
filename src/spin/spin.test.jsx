/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import React from 'react';
import { mount } from 'enzyme';

import Spin from './spin';

describe('spin', () => {
    it('should render without problems', () => {
        let spin = mount(<Spin />);

        expect(spin).toMatchSnapshot();
    });

    it('should set class `spin_visible` when visible=true', () => {
        let spin = mount(<Spin visible={ true } />);

        expect(spin.getDOMNode().className).toContain('spin_visible');
    });
});
