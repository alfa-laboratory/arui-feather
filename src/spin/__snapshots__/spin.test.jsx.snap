// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`spin should render without problems 1`] = `
<Spin
  size="m"
  visible={false}
>
  <span
    className="spin spin_size_m spin_theme_alfa-on-white"
  />
</Spin>
`;
